apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/TestLogger.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using Google.Cloud.Spanner.V1.Logging;
+using Xunit.Abstractions;
+
+namespace Google.Cloud.Spanner.Data.IntegrationTests
+{
+    internal class TestLogger : DefaultLogger
+    {
+        private static Lazy<TestLogger> s_Instance = new Lazy<TestLogger>(
```

Comment:
I believe the singleton will only get created when the class is referenced the first time. Also, since it has no instance fields, it only takes up the heap metadata header's worth of space. It doesn't seem like the savings from avoiding a creation are worth the extra code, but that's just me (actually, the `Lazy<T>` instance itself almost certainly take up more space since it is also a reference type). If you disagree that's fine.